diff --git a/fluid/of10/of10action.hh b/fluid/of10/of10action.hh
index 7c2ea40..b79e00b 100644
--- a/fluid/of10/of10action.hh
+++ b/fluid/of10/of10action.hh
@@ -219,7 +219,7 @@ public:
     virtual SetTPSrcAction* clone() {
         return new SetTPSrcAction(*this);
     }
-    IPAddress tp_port() {
+    uint16_t tp_port() {
         return this->tp_port_;
     }
     void tp_port(uint16_t tp_port) {
@@ -241,7 +241,7 @@ public:
     virtual SetTPDstAction* clone() {
         return new SetTPDstAction(*this);
     }
-    IPAddress tp_port() {
+    uint16_t tp_port() {
         return this->tp_port_;
     }
     void tp_port(uint16_t tp_port) {
diff --git a/fluid/of13/of13match.hh b/fluid/of13/of13match.hh
index 407308f..23a5f85 100644
--- a/fluid/of13/of13match.hh
+++ b/fluid/of13/of13match.hh
@@ -1113,6 +1113,12 @@ public:
     void value(uint64_t value) {
         this->value_ = value;
     }
+    uint64_t mask() const {
+        return this->mask_;
+    }
+    void mask(uint64_t mask) {
+        this->mask_ = mask;
+    }
 };
 
 class IPv6Exthdr: public OXMTLV {
diff --git a/fluid/of13/openflow-13.h b/fluid/of13/openflow-13.h
index e810fde..90c94c4 100644
--- a/fluid/of13/openflow-13.h
+++ b/fluid/of13/openflow-13.h
@@ -1380,7 +1380,7 @@ struct ofp_packet_out {
 };
 OFP_ASSERT(sizeof(struct ofp_packet_out) == 24);
 
-#define OFP_NO_BUFFER 0xffffffff
+const uint32_t OFP_NO_BUFFER = 0xffffffff;
 
 /* Packet received on port (datapath -> controller). */
 struct ofp_packet_in {
diff --git a/fluid/util/util.h b/fluid/util/util.h
index 60f34b9..17d853b 100644
--- a/fluid/util/util.h
+++ b/fluid/util/util.h
@@ -65,7 +65,7 @@
 
 #define NO_RETURN __attribute__((__noreturn__))
 #define UNUSED __attribute__((__unused__))
-#define PACKED __attribute__((__packed__))
+//#define PACKED __attribute__((__packed__))
 #define PRINTF_FORMAT(FMT, ARG1) __attribute__((__format__(printf, FMT, ARG1)))
 #define STRFTIME_FORMAT(FMT) __attribute__((__format__(__strftime__, FMT, 0)))
 #define MALLOC_LIKE __attribute__((__malloc__))
