diff --git a/pox/core.py b/pox/core.py
index b41133d..8bade69 100644
--- a/pox/core.py
+++ b/pox/core.py
@@ -271,14 +271,14 @@ class POXCore (EventMixin):
     """
     Shut down POX.
     """
-    import threading
-    if (self.starting_up or
-        threading.current_thread() is self.scheduler._thread):
-      t = threading.Thread(target=self._quit)
-      t.daemon = True
-      t.start()
-    else:
-      self._quit()
+    #import threading
+    #if (self.starting_up or
+    #    threading.current_thread() is self.scheduler._thread):
+    #  t = threading.Thread(target=self._quit)
+    #  t.daemon = True
+    #  t.start()
+    #else:
+    #  self._quit()
 
   def _quit (self):
     # Should probably do locking here
diff --git a/pox/openflow/of_01.py b/pox/openflow/of_01.py
index 6572b4c..04895e1 100644
--- a/pox/openflow/of_01.py
+++ b/pox/openflow/of_01.py
@@ -352,7 +352,7 @@ class HandshakeOpenFlowHandlers (OpenFlowHandlers):
     con.send(self._barrier)
 
     # To support old versions of cbench, just finish connecting here.
-    #self._finish_connecting(con)
+    self._finish_connecting(con)
 
   def _finish_connecting (self, con):
     con.ofnexus._connect(con)
