diff --git a/pycuda/compiler.py b/pycuda/compiler.py
index c0b5ba6..ac08c6d 100644
--- a/pycuda/compiler.py
+++ b/pycuda/compiler.py
@@ -177,7 +177,8 @@ def _get_per_user_string():
 
 def _find_pycuda_include_path():
     from pkg_resources import Requirement, resource_filename
-    return resource_filename(Requirement.parse("pycuda"), "pycuda/cuda")
+    return "pycuda/cuda" #resource_filename(Requirement.parse("pycuda"), "pycuda/cuda")
+    #return "/home/ivey/dce-python-sdn/source/ns-3-dce/files-0/pycuda/cuda" #resource_filename(Requirement.parse("pycuda"), "pycuda/cuda")
 
 
 import os
diff --git a/pycuda/curandom.py b/pycuda/curandom.py
index e1c6842..7e449fd 100644
--- a/pycuda/curandom.py
+++ b/pycuda/curandom.py
@@ -249,7 +249,7 @@ def rand(shape, dtype=np.float32, stream=None):
 # {{{ CURAND wrapper
 
 try:
-    import pycuda._driver as _curand # used to be separate module
+    import _cudadriver as _curand # used to be separate module
 except ImportError:
     def get_curand_version():
         return None
diff --git a/pycuda/driver.py b/pycuda/driver.py
index 3fa4c5b..eef71da 100644
--- a/pycuda/driver.py
+++ b/pycuda/driver.py
@@ -2,7 +2,7 @@ from __future__ import absolute_import
 from __future__ import print_function
 import six
 try:
-    from pycuda._driver import *  # noqa
+    from _cudadriver import *  # noqa
 except ImportError as e:
     if "_v2" in str(e):
         from warnings import warn
@@ -161,7 +161,7 @@ def _add_functionality():
                     arg_data.append(int(gpudata))
                     format += "P"
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         return handlers, pack(format, *arg_data)
 
     # {{{ pre-CUDA 4 call interface (stateful)
@@ -260,7 +260,7 @@ def _add_functionality():
             else:
                 func.arg_format += np.dtype(np.uintp).char
 
-        from pycuda._pvt_struct import calcsize
+        from _pvt_struct import calcsize
         func._param_set_size(calcsize(func.arg_format))
 
         return func
@@ -282,7 +282,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         func._param_setv(0, pack(func.arg_format, *args))
 
         for texref in func.texrefs:
@@ -308,7 +308,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         func._param_setv(0, pack(func.arg_format, *args))
 
         for texref in func.texrefs:
@@ -347,7 +347,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         func._param_setv(0, pack(func.arg_format, *args))
 
         for texref in func.texrefs:
@@ -457,7 +457,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         arg_buf = pack(func.arg_format, *args)
 
         for texref in func.texrefs:
@@ -471,7 +471,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         arg_buf = pack(func.arg_format, *args)
 
         for texref in func.texrefs:
@@ -507,7 +507,7 @@ def _add_functionality():
             raise TypeError("unknown keyword arguments: "
                     + ", ".join(six.iterkeys(kwargs)))
 
-        from pycuda._pvt_struct import pack
+        from _pvt_struct import pack
         arg_buf = pack(func.arg_format, *args)
 
         for texref in func.texrefs:
diff --git a/pycuda/gl/__init__.py b/pycuda/gl/__init__.py
index 56c0e64..66d9c89 100644
--- a/pycuda/gl/__init__.py
+++ b/pycuda/gl/__init__.py
@@ -1,5 +1,5 @@
 from __future__ import absolute_import
-import pycuda._driver as _drv
+import _cudadriver as _drv
 
 if not _drv.have_gl_ext(): 
     raise ImportError("PyCUDA was compiled without GL extension support")
diff --git a/pycuda/tools.py b/pycuda/tools.py
index 7a5fdce..2efdac0 100644
--- a/pycuda/tools.py
+++ b/pycuda/tools.py
@@ -33,7 +33,7 @@ OTHER DEALINGS IN THE SOFTWARE.
 
 import pycuda.driver as cuda
 from decorator import decorator
-import pycuda._driver as _drv
+import _cudadriver as _drv
 import numpy as np
 
 
diff --git a/setup.py b/setup.py
index 001ed43..7e72206 100644
--- a/setup.py
+++ b/setup.py
@@ -205,7 +205,7 @@ def main():
 
             ext_package="pycuda",
             ext_modules=[
-                NumpyExtension("_driver",
+                NumpyExtension("_cudadriver",
                     [
                         "src/cpp/cuda.cpp",
                         "src/cpp/bitlog.cpp",
diff --git a/src/cpp/mempool.hpp b/src/cpp/mempool.hpp
index be88f13..1c7b79b 100644
--- a/src/cpp/mempool.hpp
+++ b/src/cpp/mempool.hpp
@@ -14,8 +14,8 @@
 #include <boost/format.hpp>
 #include "bitlog.hpp"
 
-
-
+#define PYGPU_PACKAGE pycuda
+#define PYGPU_PYCUDA 1
 
 namespace PYGPU_PACKAGE
 {
diff --git a/src/wrapper/_pvt_struct_v2.cpp b/src/wrapper/_pvt_struct_v2.cpp
index dac3484..2834d6a 100644
--- a/src/wrapper/_pvt_struct_v2.cpp
+++ b/src/wrapper/_pvt_struct_v2.cpp
@@ -13,7 +13,8 @@
 #include "structseq.h"
 #include "structmember.h"
 #include <ctype.h>
-#include "numpy_init.hpp"
+#include <numpy/arrayobject.h>
+//#include "numpy_init.hpp"
 
 // static PyTypeObject PyStructType;
 
@@ -1576,6 +1577,8 @@ init_pvt_struct(void)
 	if (ver == NULL)
 		return;
 
+	import_array1();
+
 	m = Py_InitModule3("_pvt_struct", module_functions, module_doc);
 	if (m == NULL)
 		return;
diff --git a/src/wrapper/tools.hpp b/src/wrapper/tools.hpp
index f0fd571..b126f08 100644
--- a/src/wrapper/tools.hpp
+++ b/src/wrapper/tools.hpp
@@ -7,7 +7,8 @@
 #include <cuda.hpp>
 #include <boost/python.hpp>
 #include <numeric>
-#include "numpy_init.hpp"
+#include <numpy/arrayobject.h>
+//#include "numpy_init.hpp"
 
 
 
diff --git a/src/wrapper/wrap_cudadrv.cpp b/src/wrapper/wrap_cudadrv.cpp
index 7217597..95e39e2 100644
--- a/src/wrapper/wrap_cudadrv.cpp
+++ b/src/wrapper/wrap_cudadrv.cpp
@@ -528,8 +528,10 @@ void pycuda_expose_curand();
 
 
 
-BOOST_PYTHON_MODULE(_driver)
+BOOST_PYTHON_MODULE(_cudadriver)
 {
+  import_array();
+
   py::def("get_version", cuda_version);
 #if CUDAPP_CUDA_VERSION >= 2020
   py::def("get_driver_version", pycuda::get_driver_version);
@@ -538,7 +540,7 @@ BOOST_PYTHON_MODULE(_driver)
   // {{{ exceptions
 
 #define DECLARE_EXC(NAME, BASE) \
-  Cuda##NAME = py::handle<>(PyErr_NewException("pycuda._driver." #NAME, BASE, NULL)); \
+  Cuda##NAME = py::handle<>(PyErr_NewException("_cudadriver." #NAME, BASE, NULL)); \
   py::scope().attr(#NAME) = Cuda##NAME;
 
   {
